packages/framework-core/src/services/graphql/graphql-type-informer.ts

```diff
@@ -57,6 +57,11 @@ export class GraphQLTypeInformer {
         fields[prop.name] = {
           type: GraphQLList(graphQLPropType),
         }
+        // TODO: awaiting update of metadata-booster package to access .isArray boolean
+      } else if ((prop.typeInfo as any).isArray) {
+        fields[prop.name] = {
+          type: GraphQLList(this.getGraphQLTypeFor(prop.typeInfo.type)),
+        }
```

Comment:
If we manage to do the change that @alvaroloes suggested [here](https://github.com/boostercloud/metadata-booster/pull/1#pullrequestreview-698995075) the `if` could be simplified to a simple name check

Reply:
It seems there was already an array check above the code block I've added. However, it behaves differently the first time a type enters the if block because the metadata isn't generated yet. It causes the generated GraphQL type to be `JSONObject` for the first reference. 

Not sure if the proposed change is the best solution, feedback is welcome!